backend/lib/deal-observer.js

```diff
@@ -1,21 +1,40 @@
 /** @import {Queryable} from '@filecoin-station/deal-observer-db' */
 /** @import { Static } from '@sinclair/typebox' */
 
-import { getActorEvents, getActorEventsFilter } from './rpc-service/service.js'
+import { getActorEvents, getActorEventsFilter, getChainHead } from './rpc-service/service.js'
 import { ActiveDealDbEntry } from '@filecoin-station/deal-observer-db/lib/types.js'
 import { Value } from '@sinclair/typebox/value'
 import { convertBlockEventToActiveDealDbEntry } from './utils.js'
 
 /**
- * @param {number} blockHeight
  * @param {Queryable} pgPool
- * @param {(method:string,params:object) => object} makeRpcRequest
+ * @param {number} maxPastEpochs
+ * @param {number} finalityEpochs
+ * @param {(method:string,params:any[]) => Promise<any>} makeRpcRequest
  * @returns {Promise<void>}
  */
-export async function observeBuiltinActorEvents (blockHeight, pgPool, makeRpcRequest) {
+export const observeBuiltinActorEvents = async (pgPool, makeRpcRequest, maxPastEpochs, finalityEpochs) => {
```

Comment:
The arguments order in the comment is wrong, `maxRpcRequest` is the 2nd not the 4th argument